### DISCUSSION TITLE
GoTo specific move number

### DISCUSSION THREAD

**Discussion by RasmusRow** [mliebelt/pgn-viewer#481] [Q&A]
Hi,

I can't seem to figure out if it's possible to programmatically forward/rewind the game to a specific move number. 

Just like the notation already provided in the library, but I'd like to be able to implement some custom functionality that allows to me to click a button somewhere on the page that triggers the same event as when I click on a move number in the notations.

I hope it makes sense :) 

Thank you for your work!

**Comment by mliebelt** [comment]
You are right, there is no API to do something like that. I checked the API again, at the moment, you can only do manual moves from one position to the next. 

So let me write down the requirement: The user should be able to jump to any position in the game denoted by the move number. The move number is denoted by the array index of the half moves done. In the game `e4 e5 Nf3 Nc6` the move numbers are: `1 e4 2 e5 3 Nf3 4 Nc6`.

What about variants? We cannot say which move number will be which move in which variant (at least not so simple). For the moment, this would only work for the main line.

**Reply by RasmusRow** [reply]
Variations are - for me at least - not a priority as of right now. But I guess a way to "hack" my way around it would be to fake and fire a click event on the already provided notation. 

So I guess I can make it work - just wanted to make sure this was the last option. But thank you very much for your swift reply!

**Comment by mliebelt** [comment]
I have created an issue #482 that is a copy to this discussion. I will comment on that ticket then, when I will work on it.
